cloud/gcp/pubsub/topic/detectors-topics.tf

```diff
@@ -0,0 +1,91 @@
+	}
+
+}
+
+resource "signalfx_detector" "unavailable_sending_operations_ratio" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] GCP Pub/Sub Topic sending unavailable messages ratio"
+
+	program_text = <<-EOF
+		from signalfx.detectors.aperiodic import aperiodic
+		A = data('topic/send_message_operation_count', filter=filter('monitored_resource', 'pubsub_topic') and filter('response_code', 'unavailable') and ${module.filter-tags.filter_custom})${var.unavailable_sending_operations_ratio_aggregation_function}
+		B = data('topic/send_message_operation_count', filter=filter('monitored_resource', 'pubsub_topic') and ${module.filter-tags.filter_custom})${var.unavailable_sending_operations_ratio_aggregation_function}
+		signal = ((100*A)/B).${var.unavailable_sending_operations_ratio_transformation_function}(over='${var.unavailable_sending_operations_ratio_transformation_window}').publish('signal')
+		aperiodic.above_or_below_detector(signal, ${var.unavailable_sending_operations_ratio_threshold_critical}, 'above', lasting('${var.unavailable_sending_operations_ratio_aperiodic_duration}', ${var.unavailable_sending_operations_ratio_aperiodic_percentage})).publish('CRIT')
+		aperiodic.range_detector(signal, ${var.unavailable_sending_operations_ratio_threshold_warning}, ${var.unavailable_sending_operations_ratio_threshold_critical}, 'within_range', lasting('${var.unavailable_sending_operations_ratio_aperiodic_duration}', ${var.unavailable_sending_operations_ratio_aperiodic_percentage}), upper_strict=False).publish('WARN')
+	EOF
+
+	rule {
+		description           = "is too high >= ${var.unavailable_sending_operations_ratio_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.unavailable_sending_operations_ratio_disabled_critical, var.unavailable_sending_operations_ratio_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.unavailable_sending_operations_ratio_notifications_critical, var.unavailable_sending_operations_ratio_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "is too high >= ${var.unavailable_sending_operations_ratio_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.unavailable_sending_operations_ratio_disabled_warning, var.unavailable_sending_operations_ratio_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.unavailable_sending_operations_ratio_notifications_warning, var.unavailable_sending_operations_ratio_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
```

Comment:
As mentioned also in [google docs](https://cloud.google.com/pubsub/docs/monitoring#keeping_publishers_healthy), to keep publishers healthy, a detector for:
* `topic/send_request_count`, grouped by `response_code`

Reply:
added to issue !